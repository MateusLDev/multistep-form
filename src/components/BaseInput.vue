<script setup lang="ts">
interface ButtonProps {
  modelValue: string | number;
  label: string;
  placeholder: string;
}

withDefaults(defineProps<ButtonProps>(), {
  label: "",
  placeholder: "",
});
</script>

<template>
  <div class="input-wrapper">
    <label for="">{{ label }}</label>
    <input type="text" class="base-textfield" :placeholder />
    <span></span>
  </div>
</template>

<style scoped>
.input-wrapper {
  display: flex;
  flex-direction: column;
}

.base-textfield {
  color: #111827;

  font-family: "Poppins";
  font-size: 0.875rem;
  font-weight: 500;

  padding: 0.5rem 0.75rem;

  border: 1px solid #d1d5db;
  border-radius: 0.625rem;
}

.base-textfield:focus {
  outline-color: #fc6c4c;
}

.base-textfield::placeholder {
  color: #9ca3af;

  font-family: "Poppins";
  font-size: 0.875rem;
  font-weight: 400;
}

label {
  color: #111827;

  font-family: "Poppins";
  font-size: 0.875rem;
  font-weight: 500;

  margin-bottom: 0.5rem;
}
</style>
