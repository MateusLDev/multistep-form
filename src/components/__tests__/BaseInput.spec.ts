import { describe, it, expect } from "vitest";

import { mount } from "@vue/test-utils";
import BaseInput from "../BaseInput.vue";

describe("BaseInput", () => {
  it("Render with props", () => {
    const props = { label: "Teste input", placeholder: "Test placeholer" };
    const wrapper = mount(BaseInput, { props });

    const label = wrapper.find("label");
    const placeholder = wrapper.find("input").element.placeholder;

    expect(label.text()).toMatch(props.label);
    expect(placeholder).toMatch(props.placeholder);
  });

  it("Show value if it is passed by props", () => {
    const props = { modelValue: "Base modelValue" };
    const wrapper = mount(BaseInput, { props });

    const inputValue = wrapper.find("input").element.value;

    expect(inputValue).toMatch(props.modelValue);
  });

  it("Modelvalue is updating when change input value", () => {
    const props = { modelValue: "Base modelValue" };
    const wrapper = mount(BaseInput, { props });

    const newInputModel = "test new model";

    const input = wrapper.find('input')

    expect(input.element.value).toMatch(props.modelValue)

    input.element.value = newInputModel

    expect(input.element.value).toMatch(newInputModel)
  });
});
