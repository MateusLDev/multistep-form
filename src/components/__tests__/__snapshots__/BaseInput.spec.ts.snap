// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BaseInput > Render with props 1`] = `
VueWrapper {
  "__app": {
    "_component": {
      "name": "VTU_ROOT",
      "render": [Function],
      "setup": [Function],
    },
    "_container": <div
      data-v-app=""
    >
      <div
        class="input-wrapper"
        data-v-82c69faf=""
      >
        <label
          data-v-82c69faf=""
          for=""
        >
          Teste input
        </label>
        <input
          class="base-textfield"
          data-v-82c69faf=""
          placeholder="Test placeholer"
          type="text"
        />
        <span
          data-v-82c69faf=""
        />
      </div>
    </div>,
    "_context": {
      "app": [Circular],
      "components": {
        "transition": {
          "name": "transition",
        },
        "transition-group": {
          "name": "transition-group",
        },
      },
      "config": {
        "compilerOptions": {},
        "errorHandler": undefined,
        "globalProperties": {},
        "isNativeTag": [Function],
        "optionMergeStrategies": {},
        "performance": false,
        "warnHandler": undefined,
      },
      "directives": {},
      "emitsCache": WeakMap {},
      "mixins": [
        {
          "beforeCreate": [Function],
        },
      ],
      "optionsCache": WeakMap {},
      "propsCache": WeakMap {},
      "provides": {},
      "reload": [Function],
    },
    "_instance": {
      "a": null,
      "accessCache": {},
      "appContext": {
        "app": [Circular],
        "components": {
          "transition": {
            "name": "transition",
          },
          "transition-group": {
            "name": "transition-group",
          },
        },
        "config": {
          "compilerOptions": {},
          "errorHandler": undefined,
          "globalProperties": {},
          "isNativeTag": [Function],
          "optionMergeStrategies": {},
          "performance": false,
          "warnHandler": undefined,
        },
        "directives": {},
        "emitsCache": WeakMap {},
        "mixins": [
          {
            "beforeCreate": [Function],
          },
        ],
        "optionsCache": WeakMap {},
        "propsCache": WeakMap {},
        "provides": {},
        "reload": [Function],
      },
      "asyncDep": null,
      "asyncResolved": false,
      "attrs": {},
      "attrsProxy": null,
      "bc": null,
      "bm": null,
      "bu": null,
      "bum": null,
      "c": null,
      "components": null,
      "ctx": {
        "VTU_COMPONENT": {},
      },
      "da": null,
      "data": {},
      "devtoolsRawSetupState": {
        "VTU_COMPONENT": RefImpl {
          "__v_isRef": true,
          "__v_isShallow": false,
          "_rawValue": {},
          "_value": {},
          "dep": undefined,
        },
      },
      "directives": null,
      "ec": null,
      "effect": ReactiveEffect {
        "_depsLength": 5,
        "_dirtyLevel": 0,
        "_runnings": 0,
        "_shouldSchedule": false,
        "_trackId": 1,
        "active": true,
        "allowRecurse": true,
        "deps": [
          Map {
            [Circular] => 1,
          },
          Map {
            [Circular] => 1,
          },
          Map {
            [Circular] => 1,
          },
          Map {
            [Circular] => 1,
          },
          Map {
            [Circular] => 1,
          },
        ],
        "fn": [Function],
        "onTrack": undefined,
        "onTrigger": undefined,
        "scheduler": [Function],
        "trigger": [Function],
      },
      "emit": [Function],
      "emitsOptions": null,
      "emitted": null,
      "exposeProxy": null,
      "exposed": null,
      "inheritAttrs": undefined,
      "isDeactivated": false,
      "isMounted": true,
      "isUnmounted": false,
      "m": null,
      "next": null,
      "parent": null,
      "props": {},
      "propsDefaults": {},
      "propsOptions": [
        {},
        [],
      ],
      "provides": {},
      "proxy": {
        "VTU_COMPONENT": {},
      },
      "refs": {
        "VTU_COMPONENT": {},
      },
      "render": [Function],
      "renderCache": [],
      "root": [Circular],
      "rtc": null,
      "rtg": null,
      "scope": EffectScope {
        "_active": true,
        "cleanups": [],
        "detached": true,
        "effects": [
          ReactiveEffect {
            "_depsLength": 5,
            "_dirtyLevel": 0,
            "_runnings": 0,
            "_shouldSchedule": false,
            "_trackId": 1,
            "active": true,
            "allowRecurse": true,
            "deps": [
              Map {
                [Circular] => 1,
              },
              Map {
                [Circular] => 1,
              },
              Map {
                [Circular] => 1,
              },
              Map {
                [Circular] => 1,
              },
              Map {
                [Circular] => 1,
              },
            ],
            "fn": [Function],
            "onTrack": undefined,
            "onTrigger": undefined,
            "scheduler": [Function],
            "trigger": [Function],
          },
        ],
        "parent": undefined,
      },
      "setupContext": null,
      "setupState": {
        "VTU_COMPONENT": {},
      },
      "slots": {},
      "slotsProxy": null,
      "sp": null,
      "subTree": {
        "__v_isVNode": true,
        "__v_skip": true,
        "anchor": null,
        "appContext": null,
        "children": null,
        "component": {
          "a": null,
          "accessCache": {},
          "appContext": {
            "app": [Circular],
            "components": {
              "transition": {
                "name": "transition",
              },
              "transition-group": {
                "name": "transition-group",
              },
            },
            "config": {
              "compilerOptions": {},
              "errorHandler": undefined,
              "globalProperties": {},
              "isNativeTag": [Function],
              "optionMergeStrategies": {},
              "performance": false,
              "warnHandler": undefined,
            },
            "directives": {},
            "emitsCache": WeakMap {},
            "mixins": [
              {
                "beforeCreate": [Function],
              },
            ],
            "optionsCache": WeakMap {},
            "propsCache": WeakMap {},
            "provides": {},
            "reload": [Function],
          },
          "asyncDep": null,
          "asyncResolved": false,
          "attrs": {},
          "attrsProxy": null,
          "bc": null,
          "bm": null,
          "bu": null,
          "bum": null,
          "c": null,
          "components": {},
          "ctx": {
            "label": "Teste input",
            "modelValue": undefined,
            "placeholder": "Test placeholer",
          },
          "da": null,
          "data": {},
          "devtoolsRawSetupState": {
            "inputModel": RefImpl {
              "__v_isRef": true,
              "__v_isShallow": false,
              "_rawValue": undefined,
              "_value": undefined,
              "dep": Map {
                ReactiveEffect {
                  "_depsLength": 3,
                  "_dirtyLevel": 0,
                  "_runnings": 0,
                  "_shouldSchedule": false,
                  "_trackId": 1,
                  "active": true,
                  "allowRecurse": true,
                  "deps": [
                    Map {
                      [Circular] => 1,
                    },
                    Map {
                      [Circular] => 1,
                    },
                    [Circular],
                  ],
                  "fn": [Function],
                  "onTrack": undefined,
                  "onTrigger": undefined,
                  "scheduler": [Function],
                  "trigger": [Function],
                } => 1,
              },
            },
            "props": {
              "label": "Teste input",
              "modelValue": undefined,
              "placeholder": "Test placeholer",
            },
          },
          "directives": null,
          "ec": null,
          "effect": ReactiveEffect {
            "_depsLength": 3,
            "_dirtyLevel": 0,
            "_runnings": 0,
            "_shouldSchedule": false,
            "_trackId": 1,
            "active": true,
            "allowRecurse": true,
            "deps": [
              Map {
                [Circular] => 1,
              },
              Map {
                [Circular] => 1,
              },
              Map {
                [Circular] => 1,
              },
            ],
            "fn": [Function],
            "onTrack": undefined,
            "onTrigger": undefined,
            "scheduler": [Function],
            "trigger": [Function],
          },
          "emit": [Function],
          "emitsOptions": {
            "update:modelValue": null,
          },
          "emitted": null,
          "exposeProxy": {},
          "exposed": {},
          "inheritAttrs": undefined,
          "isDeactivated": false,
          "isMounted": true,
          "isUnmounted": false,
          "m": null,
          "next": null,
          "parent": [Circular],
          "props": {
            "label": "Teste input",
            "modelValue": undefined,
            "placeholder": "Test placeholer",
          },
          "propsDefaults": {},
          "propsOptions": [
            {
              "label": {
                "0": false,
                "1": true,
                "default": "",
                "required": true,
                "type": [Function],
              },
              "modelValue": {
                "0": false,
                "1": true,
                "required": false,
                "type": [
                  [Function],
                  [Function],
                ],
              },
              "placeholder": {
                "0": false,
                "1": true,
                "default": "",
                "required": true,
                "type": [Function],
              },
            },
            [
              "label",
              "placeholder",
            ],
          ],
          "provides": {},
          "proxy": {
            "label": "Teste input",
            "modelValue": undefined,
            "placeholder": "Test placeholer",
          },
          "refs": {},
          "render": [Function],
          "renderCache": [
            [Function],
            [Function],
          ],
          "root": [Circular],
          "rtc": null,
          "rtg": null,
          "scope": EffectScope {
            "_active": true,
            "cleanups": [],
            "detached": true,
            "effects": [
              ReactiveEffect {
                "_depsLength": 3,
                "_dirtyLevel": 0,
                "_runnings": 0,
                "_shouldSchedule": false,
                "_trackId": 1,
                "active": true,
                "allowRecurse": true,
                "deps": [
                  Map {
                    [Circular] => 1,
                  },
                  Map {
                    [Circular] => 1,
                  },
                  Map {
                    [Circular] => 1,
                  },
                ],
                "fn": [Function],
                "onTrack": undefined,
                "onTrigger": undefined,
                "scheduler": [Function],
                "trigger": [Function],
              },
            ],
            "parent": undefined,
          },
          "setupContext": {
            "attrs": {},
            "emit": [Function],
            "expose": [Function],
            "slots": {},
          },
          "setupState": {
            "inputModel": undefined,
            "props": {
              "label": "Teste input",
              "modelValue": undefined,
              "placeholder": "Test placeholer",
            },
          },
          "slots": {},
          "slotsProxy": {},
          "sp": null,
          "subTree": {
            "__v_isVNode": true,
            "__v_skip": true,
            "anchor": null,
            "appContext": null,
            "children": [
              {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": null,
                "appContext": null,
                "children": "Teste input",
                "component": null,
                "ctx": [Circular],
                "dirs": null,
                "dynamicChildren": null,
                "dynamicProps": null,
                "el": <label
                  data-v-82c69faf=""
                  for=""
                >
                  Teste input
                </label>,
                "key": null,
                "patchFlag": 1,
                "props": {
                  "for": "",
                },
                "ref": null,
                "scopeId": "data-v-82c69faf",
                "shapeFlag": 9,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 0,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "transition": null,
                "type": "label",
              },
              {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": null,
                "appContext": null,
                "children": null,
                "component": null,
                "ctx": [Circular],
                "dirs": [
                  {
                    "arg": undefined,
                    "dir": {
                      "beforeUpdate": [Function],
                      "created": [Function],
                      "getSSRProps": [Function],
                      "mounted": [Function],
                    },
                    "instance": {},
                    "modifiers": {},
                    "oldValue": undefined,
                    "value": undefined,
                  },
                ],
                "dynamicChildren": null,
                "dynamicProps": [
                  "placeholder",
                ],
                "el": <input
                  class="base-textfield"
                  data-v-82c69faf=""
                  placeholder="Test placeholer"
                  type="text"
                />,
                "key": null,
                "patchFlag": 40,
                "props": {
                  "class": "base-textfield",
                  "onInput": [Function],
                  "onUpdate:modelValue": [Function],
                  "placeholder": "Test placeholer",
                  "type": "text",
                },
                "ref": null,
                "scopeId": "data-v-82c69faf",
                "shapeFlag": 1,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 0,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "transition": null,
                "type": "input",
              },
              {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": null,
                "appContext": null,
                "ce": undefined,
                "children": null,
                "component": null,
                "ctx": null,
                "dirs": null,
                "dynamicChildren": null,
                "dynamicProps": null,
                "el": <span
                  data-v-82c69faf=""
                />,
                "key": null,
                "patchFlag": -1,
                "props": null,
                "ref": null,
                "scopeId": "data-v-82c69faf",
                "shapeFlag": 1,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 0,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "transition": null,
                "type": "span",
              },
            ],
            "component": null,
            "ctx": [Circular],
            "dirs": null,
            "dynamicChildren": [
              {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": null,
                "appContext": null,
                "children": "Teste input",
                "component": null,
                "ctx": [Circular],
                "dirs": null,
                "dynamicChildren": null,
                "dynamicProps": null,
                "el": <label
                  data-v-82c69faf=""
                  for=""
                >
                  Teste input
                </label>,
                "key": null,
                "patchFlag": 1,
                "props": {
                  "for": "",
                },
                "ref": null,
                "scopeId": "data-v-82c69faf",
                "shapeFlag": 9,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 0,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "transition": null,
                "type": "label",
              },
              {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": null,
                "appContext": null,
                "children": null,
                "component": null,
                "ctx": [Circular],
                "dirs": [
                  {
                    "arg": undefined,
                    "dir": {
                      "beforeUpdate": [Function],
                      "created": [Function],
                      "getSSRProps": [Function],
                      "mounted": [Function],
                    },
                    "instance": {},
                    "modifiers": {},
                    "oldValue": undefined,
                    "value": undefined,
                  },
                ],
                "dynamicChildren": null,
                "dynamicProps": [
                  "placeholder",
                ],
                "el": <input
                  class="base-textfield"
                  data-v-82c69faf=""
                  placeholder="Test placeholer"
                  type="text"
                />,
                "key": null,
                "patchFlag": 40,
                "props": {
                  "class": "base-textfield",
                  "onInput": [Function],
                  "onUpdate:modelValue": [Function],
                  "placeholder": "Test placeholer",
                  "type": "text",
                },
                "ref": null,
                "scopeId": "data-v-82c69faf",
                "shapeFlag": 1,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 0,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "transition": null,
                "type": "input",
              },
            ],
            "dynamicProps": null,
            "el": <div
              class="input-wrapper"
              data-v-82c69faf=""
            >
              <label
                data-v-82c69faf=""
                for=""
              >
                Teste input
              </label>
              <input
                class="base-textfield"
                data-v-82c69faf=""
                placeholder="Test placeholer"
                type="text"
              />
              <span
                data-v-82c69faf=""
              />
            </div>,
            "key": null,
            "patchFlag": 0,
            "props": {
              "class": "input-wrapper",
            },
            "ref": null,
            "scopeId": "data-v-82c69faf",
            "shapeFlag": 17,
            "slotScopeIds": null,
            "ssContent": null,
            "ssFallback": null,
            "staticCount": 0,
            "suspense": null,
            "target": null,
            "targetAnchor": null,
            "transition": null,
            "type": "div",
          },
          "suspense": null,
          "suspenseId": 0,
          "type": {
            "__file": "C:/Users/Weback/Desktop/Estudos/multistep-form/src/components/BaseInput.vue",
            "__name": "BaseInput",
            "__scopeId": "data-v-82c69faf",
            "components": {},
            "emits": [
              "update:modelValue",
            ],
            "props": {
              "label": {
                "default": "",
                "required": true,
                "type": [Function],
              },
              "modelValue": {
                "required": false,
                "type": [
                  [Function],
                  [Function],
                ],
              },
              "placeholder": {
                "default": "",
                "required": true,
                "type": [Function],
              },
            },
            "render": [Function],
            "setup": [Function],
          },
          "u": null,
          "uid": 1,
          "um": null,
          "update": [Function],
          "vnode": [Circular],
          "withProxy": null,
        },
        "ctx": [Circular],
        "dirs": null,
        "dynamicChildren": null,
        "dynamicProps": null,
        "el": <div
          class="input-wrapper"
          data-v-82c69faf=""
        >
          <label
            data-v-82c69faf=""
            for=""
          >
            Teste input
          </label>
          <input
            class="base-textfield"
            data-v-82c69faf=""
            placeholder="Test placeholer"
            type="text"
          />
          <span
            data-v-82c69faf=""
          />
        </div>,
        "key": null,
        "patchFlag": 0,
        "props": {
          "label": "Teste input",
          "placeholder": "Test placeholer",
          "ref": "VTU_COMPONENT",
        },
        "ref": {
          "f": false,
          "i": [Circular],
          "k": undefined,
          "r": "VTU_COMPONENT",
        },
        "scopeId": null,
        "shapeFlag": 4,
        "slotScopeIds": null,
        "ssContent": null,
        "ssFallback": null,
        "staticCount": 0,
        "suspense": null,
        "target": null,
        "targetAnchor": null,
        "transition": null,
        "type": {
          "__file": "C:/Users/Weback/Desktop/Estudos/multistep-form/src/components/BaseInput.vue",
          "__name": "BaseInput",
          "__scopeId": "data-v-82c69faf",
          "components": {},
          "emits": [
            "update:modelValue",
          ],
          "props": {
            "label": {
              "default": "",
              "required": true,
              "type": [Function],
            },
            "modelValue": {
              "required": false,
              "type": [
                [Function],
                [Function],
              ],
            },
            "placeholder": {
              "default": "",
              "required": true,
              "type": [Function],
            },
          },
          "render": [Function],
          "setup": [Function],
        },
      },
      "suspense": null,
      "suspenseId": 0,
      "type": {
        "name": "VTU_ROOT",
        "render": [Function],
        "setup": [Function],
      },
      "u": null,
      "uid": 0,
      "um": null,
      "update": [Function],
      "vnode": {
        "__v_isVNode": true,
        "__v_skip": true,
        "anchor": null,
        "appContext": {
          "app": [Circular],
          "components": {
            "transition": {
              "name": "transition",
            },
            "transition-group": {
              "name": "transition-group",
            },
          },
          "config": {
            "compilerOptions": {},
            "errorHandler": undefined,
            "globalProperties": {},
            "isNativeTag": [Function],
            "optionMergeStrategies": {},
            "performance": false,
            "warnHandler": undefined,
          },
          "directives": {},
          "emitsCache": WeakMap {},
          "mixins": [
            {
              "beforeCreate": [Function],
            },
          ],
          "optionsCache": WeakMap {},
          "propsCache": WeakMap {},
          "provides": {},
          "reload": [Function],
        },
        "children": null,
        "component": [Circular],
        "ctx": null,
        "dirs": null,
        "dynamicChildren": null,
        "dynamicProps": null,
        "el": <div
          class="input-wrapper"
          data-v-82c69faf=""
        >
          <label
            data-v-82c69faf=""
            for=""
          >
            Teste input
          </label>
          <input
            class="base-textfield"
            data-v-82c69faf=""
            placeholder="Test placeholer"
            type="text"
          />
          <span
            data-v-82c69faf=""
          />
        </div>,
        "key": null,
        "patchFlag": 0,
        "props": null,
        "ref": null,
        "scopeId": null,
        "shapeFlag": 4,
        "slotScopeIds": null,
        "ssContent": null,
        "ssFallback": null,
        "staticCount": 0,
        "suspense": null,
        "target": null,
        "targetAnchor": null,
        "transition": null,
        "type": {
          "name": "VTU_ROOT",
          "render": [Function],
          "setup": [Function],
        },
      },
      "withProxy": null,
    },
    "_props": null,
    "_uid": 0,
    "component": [Function],
    "config": {
      "compilerOptions": {},
      "errorHandler": undefined,
      "globalProperties": {},
      "isNativeTag": [Function],
      "optionMergeStrategies": {},
      "performance": false,
      "warnHandler": undefined,
    },
    "directive": [Function],
    "mixin": [Function],
    "mount": [Function],
    "provide": [Function],
    "runWithContext": [Function],
    "unmount": [Function],
    "use": [Function],
    "version": "3.4.21",
  },
  "__setProps": [Function],
  "cleanUpCallbacks": [
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
  ],
  "componentVM": {},
  "isDisabled": [Function],
  "rootVM": {
    "VTU_COMPONENT": {},
  },
  "wrapperElement": <div
    class="input-wrapper"
    data-v-82c69faf=""
  >
    <label
      data-v-82c69faf=""
      for=""
    >
      Teste input
    </label>
    <input
      class="base-textfield"
      data-v-82c69faf=""
      placeholder="Test placeholer"
      type="text"
    />
    <span
      data-v-82c69faf=""
    />
  </div>,
}
`;
